### PR TITLE
Adding a default docker registry mirror

### PR DESCRIPTION
**Pull request by sledigabel**
This change allows the controller to start with a specified default
docker registry mirror and avoid having to specify it in all the runner*
objects.

The change is backward compatible, if a runner has a docker registry
mirror specified, it will superseed the default one.

### CODE REVIEW

**Comment by sledigabel** [comment]
I could not add tests as I didn't see a relevant place where to put them; the runner_controller and runnerset_controller don't seem to have dedicated unit tests.
I'll be happy to add some tests if that's not the case!

**Comment by sledigabel** [comment]
I tested it on a local cluster, adding this to the chart value:
```yaml
dockerRegistryMirror: "http://example.com"
```

led to deploying a runner with this configuration: (note the arguments)
image

**Comment by sledigabel** [comment]
Finally, let me know if you want me to add something in the readme about this 😃

**Comment by mumoshu** [comment]
@sledigabel Hey! Thanks a lot for your contribution.

Would you mind adding some documentation about the new helm chart value used for configuring this?

https://github.com/actions-runner-controller/actions-runner-controller/tree/master/charts/actions-runner-controller#values

**Comment by sledigabel** [comment]
@mumoshu thanks for the pointer, I've added the missing value to the helm chart doc.
Should I bump the chart version too? I saw your comments on https://github.com/actions-runner-controller/actions-runner-controller/pull/674/files#r667400153, I'm guessing this PR is in a similar situation?

File: main.go [hunk 1]
```diff
@@ -88,6 +89,7 @@ func main() {
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&runnerImage, "runner-image", defaultRunnerImage, "The image name of self-hosted runner container.")
 	flag.StringVar(&dockerImage, "docker-image", defaultDockerImage, "The image name of docker sidecar container.")
+	flag.StringVar(&dockerRegistryMirror, "docker-registry-mirror", "", "The image name of docker sidecar container.")
```
Comment (mumoshu) [inline main.go:92]: This almost looks good to me, but this flag's description seems not good. Could you come up with a better one? 😄

Comment (sledigabel) [reply main.go:92]: 😱  how did this happen! will fix this soon. Nice spot.

**Comment by sledigabel** [comment]
@mumoshu Just updated the description. Thank you for your review!

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks a lot for your contribution @sledigabel ☺️

**Pull request merged**